presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -264,12 +265,18 @@ private void exportGroup(InternalResourceGroup group, Boolean export)
     private ResourceGroupId selectGroup(Session session)
     {
         SelectionContext context = new SelectionContext(session.getIdentity().getPrincipal().isPresent(), session.getUser(), session.getSource(), getQueryPriority(session));
-        for (ResourceGroupSelector selector : configurationManager.get().getSelectors()) {
-            Optional<ResourceGroupId> group = selector.match(context);
-            if (group.isPresent()) {
-                return group.get();
+        try {
+            for (ResourceGroupSelector selector : configurationManager.get().getSelectors()) {
+                Optional<ResourceGroupId> group = selector.match(context);
+                if (group.isPresent()) {
+                    return group.get();
+                }
             }
         }
+        catch (RuntimeException e) {
+            log.error(e, "Error matching selection rules");
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Error matching selection rules");
```

Comment:
Actually, maybe we can make this message better for the end user, when an admin misconfigures the system.  For example, "Resource group selection are invalid, and selection failed".  @electrum, thoughts?